sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -102,13 +102,23 @@ public String build(Expression expr) {
         case "FLOOR":
         case "CEIL":
         case "WIDTH_BUCKET":
+        case "SUBSTRING":
+        case "UPPER":
+        case "LOWER":
+        case "TRANSLATE":
           return visitSQLFunction(name,
             Arrays.stream(e.children()).map(c -> build(c)).toArray(String[]::new));
         case "CASE_WHEN": {
           List<String> children =
             Arrays.stream(e.children()).map(c -> build(c)).collect(Collectors.toList());
           return visitCaseWhen(children.toArray(new String[e.children().length]));
         }
+        case "TRIM":
```

Comment:
this doesn't match the doc now. why do we give up the LTRIM/RTRIM approach? I'd like to see something like
```
case "TRIM":
  visitTrim("BOTH", children...)
case "LTRIM":
  visitTrim("LEFT", childre...)
...
```